src/Storages/StorageDistributed.cpp

```diff
@@ -529,8 +529,11 @@ void StorageDistributed::read(
     const auto & modified_query_ast = rewriteSelectQuery(
         query_info.query, remote_database, remote_table, remote_table_function_ptr);
 
+    SelectQueryOptions select_query_options(processed_stage);
+    select_query_options.only_analyze = true;
+
     Block header =
-        InterpreterSelectQuery(query_info.query, context, SelectQueryOptions(processed_stage)).getSampleBlock();
+        InterpreterSelectQuery(query_info.query, context, select_query_options).getSampleBlock();
```

Comment:
There is a chaining code style around:  `SelectQueryOptions(processed_stage).analyze()`